transport.go

```diff
@@ -78,7 +78,7 @@ func addWebhookRequestDecoder(config transportConfig) kithttp.DecodeRequestFunc
 	return func(c context.Context, r *http.Request) (request interface{}, err error) {
 		requestPayload, err := ioutil.ReadAll(r.Body)
 		if err != nil {
-			return nil, err
+			return nil, &erraux.Error{Err: err, Message: "bad request body", Code: http.StatusBadRequest}
```

Comment:
I could be wrong but I think this could also happen due to network errors which is not a bad user request

Reply:
I had a hard time finding more specific documentation but here's one SOF question https://stackoverflow.com/questions/30004945/why-couldnt-go-read-a-request-properly. 

The gist of it is that we should probably not assume that failing to read the raw body bytes is a user error. I think if the json unmarshal fails (line 89), we could safely assume the user had an error in their JSON and we could safely return a 400 status code.

Reply:
I've reverted this line to just return the err again